src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexAhoCorasick.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// The RegexAhoCorasick object once ready, runs the Aho-Corasick algorithm to find multiple string matches in the pattern efficiently.
+
+using System.Collections.Generic;
+
+namespace System.Text.RegularExpressions
+{
+    internal class TrieNode
+    {
+        public TrieNode()
+        {
+            Children = new Dictionary<char, int>();
```

Comment:
I am guessing a Dictionary is pretty heavyweight in most cases. There are various implementations floating around of dictionaries that start of as lists.